akka-http-core/src/main/scala/akka/http/impl/model/parser/HeaderParser.scala

```diff
@@ -10,10 +10,12 @@ import akka.parboiled2._
 import akka.shapeless._
 import akka.http.scaladsl.model._
 
+private[http] case class HeaderParserSettings(uriParsingMode: Uri.ParsingMode = Uri.ParsingMode.Relaxed)
```

Comment:
Ah looks OK then. Just saw the default value in `HeaderParser` but overlooked that they are explicitly reused from the parser settings in `HttpHeaderParser`.